Allow extra args for all helm commands: Install, Upgrade, Delete, and Rollback

ndhanushkodi opened the pull request:
This addresses @yorinasub17 's comment on my previous PR that ExtraArgs should be for every command, and implements their suggestion to have a map from subcommand to a list of args.

### GH Issue
[GH Issue 731](https://github.com/gruntwork-io/terratest/issues/731)
[GH Issue 726](https://github.com/gruntwork-io/terratest/issues/726)

### Changes Proposed
This PR allows passing arguments to the helm command when using the `Install`, `Upgrade`, `Delete`, and `Rollback` functions in the `helm` module.

### Test output
The test output can be found in this [gist](https://gist.github.com/ndhanushkodi/12e4c428b2f2d48d78decdb1eda00e67)

Closes #731

yorinasub17 commented on modules/helm/options.go:
```diff
@@ -15,5 +15,5 @@ type Options struct {
 	Version        string              // Version of chart
 	Logger         *logger.Logger      // Set a non-default logger that should be used. See the logger package for more info.
-	ExtraArgs      []string            // Extra arguments to pass to the helm install/upgrade command
+	ExtraArgs      map[string][]string // Extra arguments to pass to the helm install/upgrade/rollback/delete command
```
```suggestion
	ExtraArgs      map[string][]string // Extra arguments to pass to the helm install/upgrade/rollback/delete command. The key signals the command (e.g., install) while the values are the extra arguments to pass through.
```

ndhanushkodi replied:
Thanks for the suggestion, I made the change! Resolving this.

yorinasub17 reviewed (commented):
Thanks for the contribution! This LGTM! Had one minor comment suggestion. Will kick off a build shortly.

ndhanushkodi commented:
Thanks for the review @yorinasub17, I just addressed the comments, and let me know if there's anything else you'd like me to do before this is merged. Thanks!

yorinasub17 reviewed (approved):
Update LGTM and helm tests passed, so this is good to go! Will merge now. Thanks again for your contribution!
